cmd/maya-apiserver/spc-watcher/handler.go

```diff
@@ -143,40 +140,140 @@ func (c *Controller) getSpcResource(key string) (*apis.StoragePoolClaim, error)
 	return spcGot, nil
 }
 
+// synSpc is the function which tries to converge to a desired state for the spc.
 func (c *Controller) syncSpc(spcGot *apis.StoragePoolClaim) error {
-	if len(spcGot.Spec.Disks.DiskList) > 0 {
-		// TODO : reconciliation for manual storagepool provisioning
-		glog.V(1).Infof("No reconciliation needed for manual provisioned pool of storagepoolclaim %s", spcGot.Name)
-		return nil
+	glog.V(1).Infof("Reconciling storagepoolclaim %s", spcGot.Name)
+	err := c.addEventHandler(spcGot, true)
+	return err
+}
+
+// addEventHandler is the event handler for the add event of spc.
+func (c *Controller) addEventHandler(spc *apis.StoragePoolClaim, resync bool) error {
+	mutateSpc, err := validate(spc)
+	if err != nil {
+		return err
 	}
-	glog.V(1).Infof("Syncing storagepoolclaim %s", spcGot.Name)
-	// Get kubernetes clientset
-	// namespaces is not required, hence passed empty.
-	newK8sClient, err := k8s.NewK8sClient("")
+	// validate can mutate spc object -- for example if maxPool field is not present in case
+	// of auto provisioning, maxPool will default to 3.
+	// We need to immediately patch SPC object here.
+	if mutateSpc && !resync {
```

Comment:
Will this case cause any issue with the following scenario:
- maya-apiserver (cstor-operator) is down
- SPC yaml is applied. 

What does resync variable indicate?